semgrep/semgrep/config_resolver.py

```diff
@@ -35,16 +38,8 @@
 
 logger = logging.getLogger(__name__)
 
-IN_DOCKER = "SEMGREP_IN_DOCKER" in os.environ
-IN_GH_ACTION = "GITHUB_WORKSPACE" in os.environ
-
-SRC_DIRECTORY = Path(os.environ.get("SEMGREP_SRC_DIRECTORY", Path("/") / "src"))
-OLD_SRC_DIRECTORY = Path("/") / "home" / "repo"
-
 RULES_REGISTRY = {"r2c": "https://semgrep.dev/c/p/r2c"}
 
-MISSING_RULE_ID = "no-rule-id"
```

Comment:
I think this was originally here because we didn't have an explicit schema. Since `semgrep.rule.Rule` specifies `id` as `str` instead of `Optional[str]`, and an `id` is required by the JSON schema I think we're safe to remove this now.